src/openrct2-ui/windows/Map.cpp

```diff
@@ -1389,13 +1389,12 @@ static void map_window_increase_map_size()
  */
 static void map_window_decrease_map_size()
 {
-    if (gMapSize < 16)
-    {
-        context_show_error(STR_CANT_DECREASE_MAP_SIZE_ANY_FURTHER, STR_NONE, {});
+    auto decreaseMapSizeAction = ChangeMapSizeAction(-1);
+    auto query = GameActions::Query(&decreaseMapSizeAction);
+    if (query->Error != GameActions::Status::Ok)
         return;
-    }
 
-    gMapSize--;
+    GameActions::Execute(&decreaseMapSizeAction);
     map_remove_out_of_range_elements();
```

Comment:
All of this code should be in the game action. You can't just decrease/increase the map size global and not actually move the boundary.